test/action_test.dart

```diff
@@ -37,14 +37,13 @@ void main() {
 
     test('should support dispatch without a payload', () async {
       Completer c = Completer();
-      Action<String> _action = Action<String>();
-
-      _action.listen((String payload) {
-        expect(payload, equals(null));
-        c.complete();
-      });
+      Action<String> _action = Action<String>()
+        ..listen((String payload) {
+          expect(payload, equals(null));
+          c.complete();
+        });
 
-      _action();
+      await _action();
       return c.future;
```

Comment:
#nit Unrelated to your changes, but these tests should probably be updated to use `expectAsync1` instead of a `Completer`. Something like:

```dart
final _action = Action<String>()
  ..listen(expectAsync1((payload) {
    expect(payload, isNull);
  });
  await _action();
```